src/scripts/main.js

```diff
@@ -1,3 +1,158 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+const headers = [...table.querySelectorAll('th')];
+const tbody = table.querySelector('tbody');
+const rows = [...tbody.querySelectorAll('tr')];
+let lastSortedColumn = null;
+let isDescending = false;
+
+const getNumber = (n) => Number(n.replace(/[^0-9.-]+/g, ''));
+
+headers.forEach((header, i) => {
+  header.addEventListener('click', () => {
+    if (lastSortedColumn === i) {
+      isDescending = !isDescending;
+    } else {
+      lastSortedColumn = i;
+      isDescending = false;
+    }
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      if (Number(getNumber(cellA))) {
```

Comment:
The condition `Number(getNumber(cellA))` might not correctly identify if a cell contains a number. Consider checking if `!isNaN(getNumber(cellA))` instead.